### PR TITLE
[12.0][ADD] helpdesk_mgmt_ticket_closed_date

### PR DESCRIPTION
**Pull request by ChristianSantamaria**
All users can modify the closed date while the ticket is not closed, in that case only managers can do it.
The closed date must always have value and must be >= at the create date

### CODE REVIEW

File: helpdesk_mgmt_ticket_closed_date/models/helpdesk_ticket.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# License LGPL-3 - See https://www.gnu.org/licenses/lgpl-3.0.html
+from odoo import models, api, _, fields
+from odoo.exceptions import ValidationError
+
+
+class HelpdeskTicket(models.Model):
+    _inherit = "helpdesk.ticket"
+
+    # default=True bug
+    closed_date_permits = fields.Boolean(
```
Comment (dalonsod) [inline helpdesk_mgmt_ticket_closed_date/models/helpdesk_ticket.py]: Better naming suggestion:
```suggestion
    closed_date_editable = fields.Boolean(
```

File: helpdesk_mgmt_ticket_closed_date/models/helpdesk_ticket.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    )
+
+    def _compute_closed_date_permits(self):
+        if (
+            self.stage_id.closed and
+            not self.env.user.has_group('helpdesk_mgmt.group_helpdesk_manager')
+        ):
+            self.closed_date_permits = False
+        else:
+            self.closed_date_permits = True
```
Comment (dalonsod) [inline helpdesk_mgmt_ticket_closed_date/models/helpdesk_ticket.py]: Code optimization, singleton error prevention, and an slightly different condition (prevent `closed_date` edition whether ticket is not closed, even if user is manager):
```suggestion
        is_manager = self.env.user.has_group('helpdesk_mgmt.group_helpdesk_manager')
        for record in self:
            record.closed_date_permits = record.stage_id.closed and is_manager
```
The proposed condition has one problem: attepting to change `closed_date` when the ticket is not closed (but it's filled due to ticket was closed before). In that case, could you check if standard unsets closed_date, or it remains filled?

File: helpdesk_mgmt_ticket_closed_date/views/helpdesk_ticket_closed_date_view.xml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<?xml version='1.0' encoding='utf-8'?>
+<odoo>
+    <record id="helpdesk_ticket_view_form_closed_date" model="ir.ui.view">
+        <field name="name">helpdesk.ticket.view.form (closed_date)</field>
+        <field name="model">helpdesk.ticket</field>
+        <field name="inherit_id" ref="helpdesk_mgmt.ticket_view_form"/>
```
Comment (dalonsod) [inline helpdesk_mgmt_ticket_closed_date/views/helpdesk_ticket_closed_date_view.xml:7]: This view removes/replaces two field attributes (`readonly` and `attrs`), so it's a little bit dangeours. In those cases, the best practice is assigning a lower priority. Could you add it?

**Review by dalonsod — CHANGES_REQUESTED** [review]
Only code review

**Comment by dalonsod** [comment]
@ChristianSantamaria take a look to the new improvements:
- constrains methods now support multiple records.
- Usage of `filtered` in order to prevent `if`s.
- Improved error message (empty `closed_date` case).
- prepare method for creation date reference (we've got `helpdesk_mgmt_ticket_start_date` addon which adds a new possible reference date).
- Improved `DESCRIPTION` text (pending `README` re-creation)

@lmiguens-solvos could you check?

**Comment by lmiguens-solvos** [comment]
> @ChristianSantamaria take a look to the new improvements:
> 
> * constrains methods now support multiple records.
> * Usage of `filtered` in order to prevent `if`s.
> * Improved error message (empty `closed_date` case).
> * prepare method for creation date reference (we've got `helpdesk_mgmt_ticket_start_date` addon which adds a new possible reference date).
> * Improved `DESCRIPTION` text (pending `README` re-creation)
> 
> @lmiguens-solvos could you check?

I tested it and it works correctly for me.
I only miss the translation of the closing date validation message.

**Comment by ChristianSantamaria** [comment]
Latest improvements:
- Creation of the README file
- Creating translations for the error message

**Review by dalonsod — APPROVED** [review]
Let's merge it!

**Comment by dalonsod** [comment]
@ChristianSantamaria despite this PR is merged, one interesting task should be preparing this addon to be donated to OCA. What do you think?

The adaptation tasks should be the following:
- license and authority in manifest file according to OCA.
- implement tests.
- remove es.po file.
- re-create README.rst file, before changing addon icon.

**Pull request merged**